libraries/AP_HAL_Linux/HAL_Linux_Class.cpp

```diff
@@ -376,6 +378,12 @@ void HAL_Linux::run(int argc, char* const argv[], Callbacks* callbacks) const
         case 'd':
             utilInstance.set_custom_defaults_path(gopt.optarg);
             break;
+        case 'a':
+            utilInstance.set_affinity_core_num(atoi(gopt.optarg));
+            break;
+        case 'f':
+            utilInstance.set_affinity_core_start_index(atoi(gopt.optarg));
+            break;
```

Comment:
utilInstance is already Linux::Util, so you can just call the linux-only methods. I wonder if start_index + core_num is better than just a single param with mask. For comparison, when ardupilot is started as a systemd service, one can configure the service to do this with the property `CPUAffinity`. And that takes a range like `CPUAffinity=2,3`. Maybe we could have the same syntax to easily adapt both solutions? See  https://www.thegeekdiary.com/how-to-set-cpu-affinity-for-systemd-process-in-centos-rhel-7/

Reply:
It's also the syntax for isolcpus kernel command line parameter

Reply:
`utilInstance`  is used in libraries/AP_HAL_Linux/HAL_Linux_Class.cpp class. So I kept them consistent with other code.
However in  /AP_HAL_Linux/Thread.cpp I switched to  `Linux::Util::from(hal.util)`

Reply:
For CPUAffinity ... 
Thank you ... I liked the idea... I have implemented a bitmask to that you can define cpu's that you want to use in cpu affiliation not necessary in sequence.